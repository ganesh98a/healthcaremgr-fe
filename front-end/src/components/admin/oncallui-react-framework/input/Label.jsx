import React from 'react'
import PropTypes from 'prop-types';
import {ARF} from "../services/ARF";
import classNames from "classnames";
import { Icon } from '@salesforce/design-system-react';
/**
 * Provide label for input field - import Label from '@oncallui/react-framework/input/Label'
 */
const Label = function(props) {
    let _this = {
        required: props.required || false,
        text: props.text || props.label || props.name || '',
        id: ARF.uniqid(props, 'label'),
        class: props.className || ""
    }
    let classes = classNames({"slds-form-element__label": true, "slds-form-element__help": !!props.errorText})
    classes += " " + _this.class;
    let iswarn = props.errorText || props.warningText;
    let color = props.errorText && "error" || "warning";
    let style = props.errorText && {color: "red"};
    return (   
        (_this.text || props.children) &&
        (<label style={{...props.style, ...style} || ""} id={_this.id} className={classes} >
            {iswarn && 
            <span style={{paddingRight:"2px"}}>
                <Icon
                    assistiveText={{ lable: 'Warning' }}
                    category="utility"
                    colorVariant={color}
                    name="warning"
                    size="x-small"
                />
                </span>
            }
            {_this.required && <abbr className="slds-required" title="required">* </abbr>}
            {_this.text}
            {props.children}
        </label>) || ""
    )
}
Label.propTypes = {
    /**
     * Set to "true" if filed is required, default is optional
     */
    required: PropTypes.bool
}
export default Label