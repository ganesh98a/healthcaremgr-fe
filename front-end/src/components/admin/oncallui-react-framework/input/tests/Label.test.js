import Label from '../Label'
import { render, screen } from '@testing-library/react';

describe('Label is rendering with attributes', () => {
    test('Label element with given "id"', () => {
        render(<Label id="test-id" />);
        let id = screen.getByText((content, element) => {
            return element.tagName.toLowerCase() === 'label' && element.id === "test-id"
        });
        expect(id).toBeTruthy();
    });
    test('Label element without "id"', () => {
        render(<Label />);
        let id = screen.getByText((content, element) => {
            //screen.debug(content);
            return element.tagName.toLowerCase() === 'label' && element.id.length > 0
        });
        expect(id).toBeTruthy();
    });
});