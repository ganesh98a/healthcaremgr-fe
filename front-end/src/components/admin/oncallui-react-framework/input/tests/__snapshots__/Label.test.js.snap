// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label is rendering with attributes 1`] = `
<label
  className="slds-form-element__label"
  id="test_id"
>
  
</label>
`;
